operations/app/src/modules/network/dns.tf

```diff
@@ -8,11 +8,11 @@ resource "azurerm_private_dns_zone" "dns_zone_private" {
 
 // This resource exists once per vnet
 module "vnet_dns" {
-  for_each = data.azurerm_virtual_network.vnet
+  for_each = toset(local.vnet_names)
```

Comment:
This was changed as in higher environments Terraform was running into a chicken and egg problem with the plan. By using the local variable, Terraform is able to anticipate the resources it needs to create before the `apply`.